docs/sphinx/dev/dev_dot_product_example.rst

```diff
@@ -0,0 +1,74 @@
+********************
+Dot Product Example
+********************
+The following example outlines how to create a dot product between two vectors: A and B, using a `kotekan` pipeline.
+
+Create a Dot Product Stage
+--------------------------
+First we need to create a `kotekan` `Stage` that consumes two input buffers, i.e. vector A and B, and computes the dot product. The `Stage` should also write the result of the computation, A.B, to an output buffer.
+
+The header and source files (`DotProduct.hpp/cpp`) are shown below:
+
+.. literalinclude:: ../../../lib/stages/ExampleDotProduct.hpp
+
+.. literalinclude:: ../../../lib/stages/ExampleDotProduct.cpp
+    :language: c++
+
+Compile Stage
+-------------
+To compile the stage add the source file to `lib/stages/CMakeLists.txt`:
+
+.. literalinclude:: ../../../lib/stages/CMakeLists.txt
+    :lines: 73-78
+    :linenos:
```

Comment:
This feels like it belongs in another section.  I'd like this page to focus on the config file, and running the pipeline only.